frontend/src/scenes/pipeline/hogfunctions/HogFunctionTest.tsx

```diff
@@ -102,39 +129,88 @@ export function HogFunctionTest(props: HogFunctionTestLogicProps): JSX.Element {
+                                                                        </>
+                                                                    }
+                                                                >
+                                                                    <span className="flex gap-2">
+                                                                        Make real HTTP requests
+                                                                        <IconInfo className="text-lg" />
+                                                                    </span>
+                                                                </Tooltip>
+                                                            }
+                                                        />
+                                                    )}
+                                                </LemonField>
+                                                <LemonDivider />
+                                                <LemonButton
+                                                    fullWidth
+                                                    onClick={loadSampleGlobals}
+                                                    loading={sampleGlobalsLoading}
+                                                    tooltip="Find the last event matching filters, and use it to populate the globals below."
+                                                >
+                                                    Fetch new event
+                                                </LemonButton>
+                                                <LemonDivider />
+                                                {savedGlobals.map(({ name, globals }, index) => (
+                                                    <div className="flex w-full justify-between" key={index}>
+                                                        <LemonButton
+                                                            key={index}
+                                                            onClick={() => setSampleGlobals(globals)}
+                                                            fullWidth
+                                                            className="flex-1"
+                                                        >
+                                                            {name}
+                                                        </LemonButton>
+                                                        <LemonButton
+                                                            size="small"
+                                                            icon={<IconX />}
+                                                            onClick={() => deleteSavedGlobals(index)}
+                                                            tooltip="Delete this saved set of globals"
```

Comment:
I think I'd opt for "save test data" or something as globals sounds a bit "internal"